### PR TITLE
Early access bug fixes

### PR DESCRIPTION
**Pull request by eli-lam**
Fixed:
- Ally death softlock 
- Ice skeleton softlock
- Ganiel and Sield's HP
- Fix title screen play button
- UI display positions
- OnDestroy bug for game ending scene
- Boss fight bug

### CODE REVIEW

File: Assets/EnableFinalBattle.cs [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+using UnityEngine;
+
+public class EnableFinalBattle : MonoBehaviour
+{
+    [SerializeField] private GameObject FinalBattleZone;
+    
+    void OnTriggerEnter()
+    {
+        FinalBattleZone.SetActive(true);
+    }
```
Comment (armandorusso) [inline Assets/EnableFinalBattle.cs:12]: What's this script for?

**Review by armandorusso — APPROVED** [review]
Looks good, good job on fixing the UI, loading scenes, and the boss bugs!

File: Assets/EnableFinalBattle.cs [hunk 2]
```diff
@@ -0,0 +1,13 @@
+﻿using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+public class EnableFinalBattle : MonoBehaviour
```
Comment (danielwiktorczyk) [inline Assets/EnableFinalBattle.cs:5]: I think `EnableFinalBattle` as a verb isn't a good idea for a class. 

Have you considered extending the CombatZone class? I feel like FinalBattleZone is a perfect candidate for  `is a combat zone`.

File: Assets/Scripts/Combat/Abilities/AbilityDescriptions/BonkDescription.asset [hunk 3]
```diff
@@ -12,5 +12,5 @@ MonoBehaviour:
   m_Script: {fileID: 11500000, guid: fd2e78d0aff466c449c00d1841840feb, type: 3}
   m_Name: BonkDescription
   m_EditorClassIdentifier: 
-  Name: Bonk
+  Name: 'Bonk - Candy Cost: 0'
```
Comment (danielwiktorczyk) [inline Assets/Scripts/Combat/Abilities/AbilityDescriptions/BonkDescription.asset:15]: Can we make the cost here dynamic? Adjusting to the actual cost? Might be overkill if it's tricky to do, but I can see a bug coming when the cost here !== cost on the ability. 

On that note, maybe decouple the cost from this description, and instead have a lil candy corn cost UI bubble that dynamically fetches the current ability's cost...?

File: Assets/Scripts/Combat/Combantants/Combatant.cs [hunk 4]
```diff
@@ -282,4 +283,19 @@ public void ShowBarsUI()
         HealthBarUIPanel.GetComponent<CanvasGroup>().alpha = 1f;
         HealthBarUIPanel.GetComponent<CanvasGroup>().blocksRaycasts = true;
     }
+
+    public void ResetHitpoints()
+    {
+        if(CurrentHealthPoints == 0 && !IsAlive)
```
Comment (danielwiktorczyk) [inline Assets/Scripts/Combat/Combantants/Combatant.cs:289]: It looks like in both cases, you're setting `CurrentHP = MaxHP`. If the intention is to make sure the player is alive, you can also state `IsAlive = true`, and play the Idle animation. I would remove the conditions entirely

File: ProjectSettings/ProjectSettings.asset [hunk 5]
```diff
@@ -12,7 +12,7 @@ PlayerSettings:
   targetDevice: 2
   useOnDemandResources: 0
   accelerometerFrequency: 60
-  companyName: DefaultCompany
+  companyName: The GameDJs
```
Comment (danielwiktorczyk) [inline ProjectSettings/ProjectSettings.asset:15]: aw ye!

**Review by danielwiktorczyk — CHANGES_REQUESTED** [review]
♟💭😎👏

**Comment by George540** [comment]
Looks good. You may make the UI board for ability explanations a bit smaller. Covers a big chunk of the screen

File: Assets/Scripts/Combat/Combantants/Combatant.cs (hunk 4 above)
Comment (armandorusso) [reply Assets/Scripts/Combat/Combantants/Combatant.cs]: Fixed! Removed the conditions entirely. However, now when this function is called, unity gives a weird warning. Everything works fine though.

**Review by George540 — APPROVED** [review]
Looks good to me! Feel free to merge

File: Assets/Scripts/Dialogue/Conversations/NPCs/Lost Tree Conversation.asset [hunk 6]
```diff
@@ -12,7 +12,7 @@ MonoBehaviour:
   m_Script: {fileID: 11500000, guid: e2b4478411a425943808b2200613323a, type: 3}
   m_Name: Lost Tree Conversation
   m_EditorClassIdentifier: 
-  CandyCornReward: 69
+  CandyCornReward: 8
```
Comment (danielwiktorczyk) [inline Assets/Scripts/Dialogue/Conversations/NPCs/Lost Tree Conversation.asset:15]: 😁

File: Assets/EnableFinalBattle.cs (hunk 1 above)
Comment (danielwiktorczyk) [reply Assets/EnableFinalBattle.cs:12]: triggering the final battle

File: Assets/Scripts/Combat/Abilities/AbilityDescriptions/BonkDescription.asset (hunk 3 above)
Comment (danielwiktorczyk) [reply Assets/Scripts/Combat/Abilities/AbilityDescriptions/BonkDescription.asset:15]: Let's revisit this haha

File: Assets/EnableFinalBattle.cs (hunk 2 above)
Comment (danielwiktorczyk) [reply Assets/EnableFinalBattle.cs:5]: I'll probably refactor this later haha

**Review by danielwiktorczyk — APPROVED** [review]
🐐

**Pull request merged**